jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -318,12 +352,33 @@ def vue_update_subset(self, *args):
             if len(self.subset_states) <= index:
                 return
             sub_states = self.subset_states[index]
+
+            # we need to push updates to subset in pixels. to do this when wcs
+            # linked, convert the new input subset parameters from sky to pix
+            wcs = None
+
+            if self.display_sky_coordinates:
+                wcs = self.app._get_wcs_from_subset(sub_states)
+
+                if wcs is not None:
+                    # convert newly entered sky coords to pixel
+                    skyregion = utils._subset_def_to_region(self.subset_types[index], sub)
+                    new = utils._get_pixregion_params_in_dict(skyregion.to_pixel(wcs))
+                    # convert previous entered sky coords to pixel
+                    new_orig_skyregion = utils._subset_def_to_region(self.subset_types[index], sub, val='orig')
+                    new_orig = utils._get_pixregion_params_in_dict(new_orig_skyregion.to_pixel(wcs))
+
             for d_att in sub:
                 if d_att["att"] == 'parent':  # Read-only
                     continue
-
-                if d_att["att"] == 'theta':  # Humans use degrees but glue uses radians
-                    d_val = np.radians(d_att["value"])
+                if self.display_sky_coordinates and (wcs is not None):
+                    d_att["value"] = new[d_att["att"]]
+                    d_att["orig"] = new_orig[d_att["att"]]
+
+                if (d_att["att"] == 'theta') and (self.display_sky_coordinates is False):
+                    # Humans use degrees but glue uses radians
+                    # We've already enforced this in wcs linking in _get_pixregion_params_in_dict
+                    d_val = (d_att["value"]*u.deg).to(u.rad).value
```

Comment:
No need for astropy unit conversion here:
```suggestion
                    d_val = np.radians(d_att["value"])
```